script/config.js

```diff
@@ -11,13 +11,14 @@ function getConfig() {
     tempFile: ''
   }
 
-  if (process.platform === 'darwin') {
+  const targetPlatform = process.env.DUGITE_DOWNLOAD_PLATFORM || process.platform;
```

Comment:
For posterity can you add some notes [here](https://github.com/desktop/dugite/blob/master/docs/setup.md) about this environment variable, what it represents and when you should set this?

Reply:
Thanks!

I did try to test `DUGITE_DOWNLOAD_PLATFORM=win32 npm install` in my OSX environment, but I got an error in test cases which use installed git binary.
I didn't notice about this because I use `npm install --ignore-scripts` and execute `script/download-git.js` by hand.

My some ideas to solve this problem are:

1. Skip some tests if the `DUGITE_DOWNLOAD_PLATFORM` environment variable is not equal to `process.platform` value.
2. We recommend to use `npm install --ignore-scripts` and `DUGITE_DOWNLOAD_PLATFORM=... node script/download-git.js` for downloading other platform binaries

Do you have any comments?

Reply:
Okay, I think there's two cases here that we should distinguish:

 - working with the source repository
 - `npm install --save dugite` to add it as a dependency for a package

Both of these will run `download-git.js`, but the first scenario is the one which runs the tests as part of the `npm run prepublish` step (this will go away once it's running on npm@5 - tracking this in https://github.com/desktop/dugite/issues/121).

```
npm WARN prepublish-on-install As of npm@5, `prepublish` scripts will run only for `npm publish`.
npm WARN prepublish-on-install (In npm@4 and previous versions, it also runs for `npm install`.)
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
```

I really like choosing the binaries for the current platform and executing tests when using the source repository, as invoking Git is a core part of the library, and hiding those makes me nervous.

I personally prefer to do `npm install` on the right platform, to verify that any native modules are compiled and correctly. But I also run a bunch of Windows and Linux VMs locally for testing, which I appreciate not everyone can do.

Can you clarify whether it's the contributing side or the packaging side that you're interested in supporting?